falco.yaml

```diff
@@ -773,6 +775,39 @@ metrics:
   convert_memory_to_mb: true
   include_empty_values: false
 
+  ###############################################
+# Falco driver selection                    #
+###############################################
+
+# [Stable] `driver_mode`
+#
+# --- [Description]
+#
+# Falco supports different driver modes for capturing syscall events. The choice
+# of driver mode can significantly impact the performance and compatibility of
+# Falco with your system. Choose the appropriate driver mode based on your
+# system's configuration and requirements.
+#
+# Available driver modes:
+# - `kmod`: Kernel Module (Kernel Module)
+# - `bpf`: eBPF (Extended Berkeley Packet Filter)
+# - `modern_bpf`: Modern eBPF (Modern Extended Berkeley Packet Filter)
+# - `nodriver`: No Driver (No driver, just for testing)
```

Comment:
I would probably rename the `nodriver` in something else. Today the `nodriver` is a workaround to load plugins with `syscall` source, but to the final user this is not clear using a word like `nodriver`, probably something like `syscall_plugin` would be more meaningful.

Reply:
The _no driver_ option is not related to plugins. It can be helpful in case of testing or running Falco in CI.
If we believe it is misleading, we can choose a more appropriate label, still expressing that Falco will not rely on a driver. 

For example, `none` might be a good alternative:

```
driver:
  kind: none
```

Alternatively, we can use a different config key:

```
driver:
  enabled: false
```

Reply:
> The no driver option is not related to plugins. It can be helpful in case of testing or running Falco in CI.

Looking at the code it seems also related to source plugins with "syscall" source

```cpp
		else if (s.options.nodriver) /* nodriver engine. */
		{
			// when opening a capture with no driver, Falco will first check
			// if a plugin is capable of generating raw events from the libscap
			// event table (including system events), and if none is found it
			// will use the nodriver engine.
			for (const auto& p: inspector->get_plugin_manager()->plugins())
			{
				if (p->caps() & CAP_SOURCING && p->id() == 0)
				{
					auto cfg = s.plugin_configs.at(p->name());
					falco_logger::log(LOG_INFO, "Opening '" + source + "' source with plugin '" + cfg->m_name + "'");
					inspector->open_plugin(cfg->m_name, cfg->m_open_params);
					return run_result::ok();
				}
			}
			falco_logger::log(LOG_INFO, "Opening '" + source + "' source with no driver\n");
			inspector->open_nodriver();
		}
```

Reply:
The no-driver option related to plugins implementing the `syscall` raw event source. Since Falco never had a per-plugin selector for opening a capture, but rather a per-source one, the logic has always been as follows:
- When enabling the `syscall` source, it is opened by using the selected driver
- When enabling any non-`syscall` source, it sis opened by using the loaded plugin implementing it
- When having a event-sourcing plugin implementing the `syscall` source itself, the priority is still given to the selected driver for non-ambiguity and backward compatibility with the previously existing behavior. In such case, if the selected driver is `no-driver`, then the plugin is used for opening the `syscall` event source. This was introduced in Falco 0.35.0

Reply:
@jasondellaluce @Andreagit97 

My point was that `--no-driver` itself is not strictly related to plugins. On the other hand, it's related to `syscall` since the `syscall` data source needed a driver historically. The fact that we need `nodriver` for loading the event-sourcing plugin implementing the syscall source is the consequence of maintaining the backward compatibility, but we can change this behavior in the future, can't we? :thinking:

Moreover, there is no actual implementation of a syscall plugin, while `nodriver` for the CI purpose is a compelling use case currently [in use](https://github.com/falcosecurity/charts/blob/master/tests/falco-test-ci.yaml#L14C8-L14C8).


So I still think the `nodriver` option is a compelling option that deserves to exist regardless of the plugin use case. Let me know if you disagree!